### PR TITLE
Backmerge from Github Enterprise

### PR DESCRIPTION
**Pull request by cd-at-iteratec**
- Added crossplane-efs chart
- Added refactored crossplane-irsa

### CODE REVIEW

File: charts/crossplane-efs/templates/xrd.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                  type: string
+                  description: "Unique id used for tagging all dependent components."
+                parameters:
+                  type: object
+                  properties:
+                    region:
+                      type: string
+                      description: "AWS Region in which to create network resources."
+                      enum: ["eu-west-1", "eu-central-1", "us-east-1"]
+                    performanceMode:
```
Comment (mithie) [inline charts/crossplane-efs/templates/xrd.yaml:38]: throughputMode and provisionedThroughputInMibps are still missing in the xrd and composition. 
can you adapt that from the current setup in the integration environment?

File: charts/crossplane-efs/values.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  # xrd.version -- The version of this component (MUST not be overridden from outside).
+  version: v1alpha1
+
+composition:
+  # composition.name -- The name of the crossplane composition (MUST not be overridden from outside).
+  efs:
+    name: crossplane-efs
+    network:
+      name: crossplane-efs-network
+      vpcId: vpc-0d4f2f4f6e1fc386c
```
Comment (mithie) [inline charts/crossplane-efs/values.yaml:17]: please remove all concrete environment specific parameters from values.yaml

**Review by mithie — CHANGES_REQUESTED** [review]

File: charts/crossplane-efs/templates/xrd.yaml (hunk 1 above)
Comment (cd-at-iteratec) [reply charts/crossplane-efs/templates/xrd.yaml:38]: Puh, I copied from master. Need to find the changes

File: charts/crossplane-efs/values.yaml (hunk 2 above)
Comment (cd-at-iteratec) [reply charts/crossplane-efs/values.yaml:17]: Ok, so clear vpcId or even more ?

Comment (mithie) [reply charts/crossplane-efs/values.yaml:17]: all ;)

File: charts/crossplane-efs/templates/xrd.yaml (hunk 1 above)
Comment (mithie) [reply charts/crossplane-efs/templates/xrd.yaml:38]: see infra deployment repo

Comment (cd-at-iteratec) [reply charts/crossplane-efs/templates/xrd.yaml:38]: done

File: charts/crossplane-efs/values.yaml (hunk 2 above)
Comment (cd-at-iteratec) [reply charts/crossplane-efs/values.yaml]: done

**Review by mithie — APPROVED** [review]

**Pull request merged**